pywinauto/controls/hwndwrapper.py

```diff
@@ -457,52 +460,147 @@ def send_message(self, message, wparam = 0, lparam = 0):
 
-        Parses modifiers Shift(+), Control(^), Menu(%) and Sequences like "{TAB}", "{Enter}"
-        For more information about Sequences and Modifiers navigate to keyboard.py
+        If a virtual key with no corresponding character is encountered
+        (e.g. VK_LEFT, VK_DELETE), a KeySequenceError is raised.
         """
-
         input_locale_id = ctypes.windll.User32.GetKeyboardLayout(0)
+        keys = keyboard.parse_keys(chars, with_spaces, with_tabs, with_newlines)
 
-        keys = keyboard.parse_keys(message, with_spaces, with_tabs, with_newlines)
         for key in keys:
+            key_info = key.get_key_info()
+            flags = key_info[2]
 
-            vk, scan, flags = key.get_key_info()
+            unicode_char = (
+                flags & keyboard.KEYEVENTF_UNICODE ==
+                keyboard.KEYEVENTF_UNICODE)
 
-            if flags & keyboard.KEYEVENTF_UNICODE == 0:
-                char = keyboard.MapVirtualKey(vk, 2)
-            else:
-                # Indicates that we actually just have a unicode codepoint
-                char = scan
-                vk = ctypes.windll.User32.VkKeyScanExW(char, input_locale_id) & 0xFF # TODO: use shift state in high order byte
+            if unicode_char:
+                _, char = key_info[:2]
+                vk = ctypes.windll.User32.VkKeyScanExW(char, input_locale_id) & 0xFF
                 scan = keyboard.MapVirtualKey(vk, 0)
-
-            if key.down and vk > 0:
-                # TODO: {CTRL} (^) modifier doesn't work
-                # + SHIFT down
-                # - CTRL down
-                # print('key', key, 'down')
-                lparam = 1 << 0 | scan << 16 | (flags & 1) << 24 | 0 << 29 | 0 << 31
-                win32api.SendMessage(self.handle, win32con.WM_KEYDOWN, vk, lparam)
+            else:
+                vk, scan = key_info[:2]
+                char = keyboard.MapVirtualKey(vk, 2)
 
             if char > 0:
                 lparam = 1 << 0 | scan << 16 | (flags & 1) << 24
                 win32api.SendMessage(self.handle, win32con.WM_CHAR, char, lparam)
+            else:
+                raise keyboard.KeySequenceError('no corresponding char for ' +
+                                                str(key))
+
+    # -----------------------------------------------------------
+    def send_keystrokes(self,
+                   keystrokes,
+                   with_spaces=True,
+                   with_tabs=True,
+                   with_newlines=True):
+        """
+        Silently send keystrokes to the control
+
+        Parses modifiers Shift(+), Control(^), Menu(%) and Sequences like "{TAB}", "{ENTER}"
+        For more information about Sequences and Modifiers navigate to keyboard.py
+        """
+        user32 = ctypes.windll.User32
+        PBYTE256 = ctypes.c_ubyte * 256
+
+        win32gui.SendMessage(self.handle, win32con.WM_ACTIVATE,
+                             win32con.WA_ACTIVE, 0)
```

Comment:
`set_focus` steals global keyboard focus and in case of Notepad moves the window to the foreground. As far as I understand, if the user is fine with the window being in the foreground, `type_keys` is the official and the most robust way to pass keys to the application. Therefore, it's important for `send_chars` and `send_keys` to keep the window inactive/minimized.

I've tried multiple ways to make an app respond to silently sending input: doing nothing but attaching a message queue, `WM_ACTIVATE`, `SetActiveWindow`, `SetFocus`, `HwndWrapper.set_focus` and `HwndWrapper.set_keyboard_focus`. For CmnCtrl3.exe these alternatives are seemingly irrelevant (they all work, except for stealing global focus with `HwndWrapper.set_focus` and a crash inside `set_keyboard_focus`). On the other hand, in Notepad the `send_keystrokes` is most functional when activating window via `WM_ACTIVATE` or `SetFocus` WinAPI call.

I understand that this approach to programming is amateurish and the current solution is hacky, but so are both `send_chars` and `send_keystrokes`, because of how inflexible Windows input mechanism is. The best that I can do is to add disclaimers to the docstrings of `send_chars` and `send_keystrokes` explaining their purpose and limitations.